fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalScanNode.java

```diff
@@ -33,10 +36,20 @@
  * For example:
  * hive, iceberg, hudi, es, odbc
  */
-public class ExternalScanNode extends ScanNode {
+public abstract class ExternalScanNode extends ScanNode {
+    private static final Logger LOG = LogManager.getLogger(ExternalScanNode.class);
 
     // set to false means this scan node does not need to check column priv.
-    private boolean needCheckColumnPriv;
+    protected boolean needCheckColumnPriv;
+
+    // For explain
+    protected long inputSplitsNum = 0;
```

Comment:
I think this 4 fields should be in `FileScanNode`?

Reply:
Will fix this in next pr